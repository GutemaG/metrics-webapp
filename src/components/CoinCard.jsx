import React from 'react';
import { Card, Col } from 'react-bootstrap';
import './style/CoinCard.css';

function CoinCard() {
  return (
    <div>
      <Col border="danger" className="m-1">
        <Card className="bg-dark p-4 coin-card">
          <Card.Header as="h2">
            <span className="coin-name text-decoration-underline m-1">BitCoin</span>
            <span className="coin-increasing text-success text-decoration-none">+</span>
          </Card.Header>
          <Card.Text className="m-1 coin-price">
            $23006.8687
          </Card.Text>
        </Card>
      </Col>
    </div>
  );
}

export default CoinCard;
