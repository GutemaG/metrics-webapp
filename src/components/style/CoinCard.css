.coin-card {
  border: 1px solid #ccc;
  border-radius: 10px;
  cursor: pointer;
  margin: 0.3rem;
  transition: all 0.5s;
  position: relative;
}

.coin-card:hover {
  transform: scale(1.1);
  border: 2px solid gold;
  z-index: 10;
}

.coin-price {
  font-weight: 600;
  font-size: 1.4rem;
}

.coin-card-header {
  display: flex;
  flex-direction: column;
  gap: 5px;
  justify-content: center;
  align-items: center;
}

.coin-card-header img {
  width: 50%;
  height: 50%;
}

.rank-badge {
  position: absolute;
  top: 0;
  right: 0;
}
