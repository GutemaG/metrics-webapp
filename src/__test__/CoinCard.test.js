import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import CoinCard from '../components/CoinCard';

describe('Testing CoinCard component', () => {
  const coin = {
    id: 'bitcoin',
    icon: 'https://static.coinstats.app/coins/1650455588819.png',
    name: 'Bitcoin',
    symbol: 'BTC',
    rank: 1,
    price: 22959.100151023245,
    priceBtc: 1,
    volume: 36394172890.744354,
    marketCap: 442766521921.6851,
    availableSupply: 19285012,
    totalSupply: 21000000,
    priceChange1h: 0.39,
    priceChange1d: -0.15,
    priceChange1w: -0.15,
    websiteUrl: 'http://www.bitcoin.org',
    twitterUrl: 'https://twitter.com/bitcoin',
  };
  const coinCard = render(
    <Provider store={store}>
      <BrowserRouter>
        <CoinCard id={coin.id} coin={coin} />
      </BrowserRouter>
    </Provider>,
  );
  test('Render coin card component correctly', () => {
    expect(coinCard).toMatchSnapshot();
  });
});
