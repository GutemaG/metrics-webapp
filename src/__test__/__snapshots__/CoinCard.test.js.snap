// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CoinCard component Render coin card component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          class="text-white"
          href="/detail/bitcoin"
        >
          <div
            border="danger"
            class="m-1 col"
          >
            <div
              class="bg-dark p-4 coin-card card"
            >
              <h2
                class="card-header"
              >
                <div
                  class="coin-card-header"
                >
                  <span
                    class="rank-badge badge bg-secondary"
                  >
                    1
                  </span>
                  <img
                    alt="coin-icon"
                    class="justify-content-center m-"
                    src="https://static.coinstats.app/coins/1650455588819.png"
                  />
                  <span
                    class="coin-name text-decoration-underline m-1"
                  >
                    Bitcoin
                  </span>
                </div>
              </h2>
              <p
                class="m-1 coin-price card-text"
              >
                $
                22959.100
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        class="text-white"
        href="/detail/bitcoin"
      >
        <div
          border="danger"
          class="m-1 col"
        >
          <div
            class="bg-dark p-4 coin-card card"
          >
            <h2
              class="card-header"
            >
              <div
                class="coin-card-header"
              >
                <span
                  class="rank-badge badge bg-secondary"
                >
                  1
                </span>
                <img
                  alt="coin-icon"
                  class="justify-content-center m-"
                  src="https://static.coinstats.app/coins/1650455588819.png"
                />
                <span
                  class="coin-name text-decoration-underline m-1"
                >
                  Bitcoin
                </span>
              </div>
            </h2>
            <p
              class="m-1 coin-price card-text"
            >
              $
              22959.100
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
